c_src/unifex/nif/unifex/payload.c

```diff
@@ -37,38 +37,40 @@ UNIFEX_TERM unifex_payload_to_term(ErlNifEnv *env, UnifexPayload *payload) {
                               enif_make_atom(env, "unifex_payload_to_term"));
 }
 
-UnifexPayload *unifex_payload_alloc(UnifexEnv *env, UnifexPayloadType type,
-                                    unsigned int size) {
-  UnifexPayload *payload = enif_alloc(sizeof(UnifexPayload));
-  payload->size = size;
-  payload->type = type;
-  payload->owned = 1;
+int unifex_payload_alloc(UnifexEnv *env, UnifexPayloadType type,
+                                    unsigned int size, UnifexPayload *payload) {
   Shmex *p_struct;
   ShmexLibResult result;
```

Comment:
This variable is named `result`, despite the fact, that it is not returned anywhere in this function, so maybe we should rename it?

Reply:
Yeah but this is because of struct name - `ShmexLibResult`. I would leave it as it is.